client/main.go

```diff
@@ -24,6 +25,83 @@ var (
 	EventsClient   events.EventsServiceClient
 )
 
+// FullClient aggregates all the grpc clients available from Tinkerbell Server
+type FullClient struct {
+	TemplateClient template.TemplateServiceClient
+	WorkflowClient workflow.WorkflowServiceClient
+	HardwareClient hardware.HardwareServiceClient
+	EventsClient   events.EventsServiceClient
+}
+
+// NewFullClientFromGlobal is a dirty hack that returns a FullClient using the
+// global variables exposed by the client package. Globals should be avoided
+// and we will deprecated them at some point replacing this function with
+// NewFullClient. If you are strating a new project please use the last one
+func NewFullClientFromGlobal() (*FullClient, error) {
+	// This is required because we use init() too often, even more in the
+	// CLI and based on where you are sometime the clients are not initialised
+	if TemplateClient == nil {
+		err := Setup()
+		if err != nil {
+			panic(err)
+		}
+	}
+	return &FullClient{
+		TemplateClient: TemplateClient,
+		WorkflowClient: WorkflowClient,
+		HardwareClient: HardwareClient,
+		EventsClient:   EventsClient,
+	}, nil
+}
+
+// NewFullClient returns a FullClient. A structure that contains all the
+// clients made available from tink-server. This is the function you should use
+// instead of NewFullClientFromGlobal that will be deprecated soon
+func NewFullClient(conn grpc.ClientConnInterface) *FullClient {
+	return &FullClient{
+		TemplateClient: template.NewTemplateServiceClient(conn),
+		WorkflowClient: workflow.NewWorkflowServiceClient(conn),
+		HardwareClient: hardware.NewHardwareServiceClient(conn),
+		EventsClient:   events.NewEventsServiceClient(conn),
+	}
+}
+
+type ConnOptions struct {
+	CertURL       string
+	GRPCAuthority string
+}
+
+func (o *ConnOptions) SetFlags(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(&o.CertURL, "tinkerbell-cert-url", "http://127.0.0.1:42114/cert", "The URL where the certificate is located")
+	flagSet.StringVar(&o.GRPCAuthority, "tinkerbell-grpc-authority", "127.0.0.1:42113", "Link to tink-server grcp api")
```

Comment:
```suggestion
	flagSet.StringVar(&o.GRPCAuthority, "tinkerbell-grpc-authority", "127.0.0.1:42113", "Link to tink-server grpc api")
```
Sigh I do this all the time